src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationapi/config/AuthorizationApiConfig.kt

```diff
@@ -81,40 +91,83 @@ class AuthorizationApiConfig(
   private val jwtCookieAuthenticationFilter: JwtCookieAuthenticationFilter,
 ) {
 
+  class ForbiddenAuthenticationProvider : AuthenticationProvider {
+    override fun authenticate(authentication: Authentication?): Authentication = throw OAuth2AuthenticationException(OAuth2Error(OAuth2ErrorCodes.ACCESS_DENIED))
+
+    override fun supports(authentication: Class<*>?): Boolean = false
```

Comment:
I don't think this AuthenticationProvider will ever be used since the `supports` function returns false. Also, not 100% certain, but think you should be able to deny access to all of these end points by just configuring a single AuthenticationProvider on all of them (so remove all of the default AuthenticationProviders) that just returns null. This would mean that it is not authenticated. That would probably result in a 401 response so we'd only need an error handler if we wanted to convert that to a 403, but not sure we'd really need to do that.